### ISSUE TITLE
Bug: Passing invalid path to VerifyPath causes irrecoverable crash

### ISSUE THREAD

**Issue by hecon5** [joyfullservice/msaccess-vcs-addin#555]
While troubleshooting #553 and testing the fix, I discovered if you somehow hand an invalid path (such as `Other.log` as the full path), `VerifyPath` will fail out and crash irrecoverably on build, even if you've already thrown a critical error.

I have a fix ready, I'll make a separate PR for this (a refactored  #363) which should be merged before we can fix #553. This fix will also have fixes for missing drives / paths as well, which may have a side benefit of more gracefully handling issues caused by backends not being available. Obviously, this won't fix the missing drive, but it will gracefully handle it.
